File: backend/internal/constants/enums.go
package constants

// DecorStyle 装修风格。
const (
	DecorStyleModern        = "Modern"
	DecorStyleChinese       = "Chinese"
	DecorStyleNordic        = "Nordic"
	DecorStyleJapanese      = "Japanese"
	DecorStyleIndustrial    = "Industrial"
	DecorStyleMediterranean = "Mediterranean"
	DecorStyleMinimalist    = "Minimalist"
)

// PhaseStatus 设计阶段状态。
const (
	PhaseStatusNotStarted = "NotStarted"
	PhaseStatusInProgress = "InProgress"
	PhaseStatusRevision   = "Revision"
	PhaseStatusApproved   = "Approved"
)

// ConstructionPhase 施工节点名称。
const (
	ConstructionPhaseDemolition     = "Demolition"
	ConstructionPhasePlumbing       = "Plumbing"
	ConstructionPhaseCarpentry      = "Carpentry"
	ConstructionPhaseTiling         = "Tiling"
	ConstructionPhasePainting       = "Painting"
	ConstructionPhaseInstallation   = "Installation"
	ConstructionPhaseSoftFurnishing = "SoftFurnishing"
)

// PurchaseStatus 材料采购状态。
const (
	PurchaseStatusNotPurchased = "NotPurchased"
	PurchaseStatusOrdered      = "Ordered"
	PurchaseStatusDelivered    = "Delivered"
	PurchaseStatusInstalled    = "Installed"
)

// BudgetCategory 预算类别。
const (
	BudgetCategoryDesign    = "Design"
	BudgetCategoryMaterial  = "Material"
	BudgetCategoryLabor     = "Labor"
	BudgetCategoryFurniture = "Furniture"
	BudgetCategoryAppliance = "Appliance"
	BudgetCategoryOther     = "Other"
)

// MaterialCategory 材料品类。
const (
	MaterialCategoryTile     = "瓷砖"
	MaterialCategoryFloor    = "地板"
	MaterialCategoryPaint    = "油漆"
	MaterialCategoryLighting = "灯具"
	MaterialCategoryBathroom = "卫浴"
	MaterialCategoryHardware = "五金"
	MaterialCategoryBoard    = "板材"
	MaterialCategoryOther    = "其他"
)

// ProjectStatus 项目状态。
const (
	ProjectStatusDesigning  = "Designing"
	ProjectStatusQuoting    = "Quoting"
	ProjectStatusInProgress = "InProgress"
	ProjectStatusCompleted  = "Completed"
	ProjectStatusArchived   = "Archived"
)

// ConstructionStatus 施工节点状态。
const (
	ConstructionStatusPending    = "Pending"
	ConstructionStatusInProgress = "InProgress"
	ConstructionStatusCompleted  = "Completed"
	ConstructionStatusDelayed    = "Delayed"
)

// AcceptanceStatus 验收状态。
const (
	AcceptanceStatusPending = "Pending"
	AcceptanceStatusPassed  = "Passed"
	AcceptanceStatusFailed  = "Failed"
)

// Role 系统角色。
const (
	RoleAdmin          = "Admin"
	RoleDesigner       = "Designer"
	RoleContractor     = "Contractor"
	RoleOwner          = "Owner"
	RoleProjectManager = "ProjectManager"
)

var (
	DecorStyles = []string{
		DecorStyleModern, DecorStyleChinese, DecorStyleNordic, DecorStyleJapanese,
		DecorStyleIndustrial, DecorStyleMediterranean, DecorStyleMinimalist,
	}
	PhaseStatuses = []string{
		PhaseStatusNotStarted, PhaseStatusInProgress, PhaseStatusRevision, PhaseStatusApproved,
	}
	ConstructionPhases = []string{
		ConstructionPhaseDemolition, ConstructionPhasePlumbing, ConstructionPhaseCarpentry,
		ConstructionPhaseTiling, ConstructionPhasePainting, ConstructionPhaseInstallation,
		ConstructionPhaseSoftFurnishing,
	}
	PurchaseStatuses = []string{
		PurchaseStatusNotPurchased, PurchaseStatusOrdered, PurchaseStatusDelivered, PurchaseStatusInstalled,
	}
	BudgetCategories = []string{
		BudgetCategoryDesign, BudgetCategoryMaterial, BudgetCategoryLabor,
		BudgetCategoryFurniture, BudgetCategoryAppliance, BudgetCategoryOther,
	}
	ProjectStatuses = []string{
		ProjectStatusDesigning, ProjectStatusQuoting, ProjectStatusInProgress, ProjectStatusCompleted, ProjectStatusArchived,
	}
	ConstructionStatuses = []string{
		ConstructionStatusPending, ConstructionStatusInProgress, ConstructionStatusCompleted, ConstructionStatusDelayed,
	}
	AcceptanceStatuses = []string{AcceptanceStatusPending, AcceptanceStatusPassed, AcceptanceStatusFailed}
	Roles              = []string{RoleAdmin, RoleDesigner, RoleContractor, RoleOwner, RoleProjectManager}
)

// Contains 判断字符串是否在集合内。
func Contains(items []string, value string) bool {
	for _, item := range items {
		if item == value {
			return true
		}
	}
	return false
}

// MaterialCategoryBudgetMap 材料品类到预算类别的映射。
// 材料从已订货推进到已交付时，按项目 + 映射后的预算类别归集实际花费。
var MaterialCategoryBudgetMap = map[string]string{
	MaterialCategoryTile:     BudgetCategoryMaterial,
	MaterialCategoryFloor:    BudgetCategoryMaterial,
	MaterialCategoryPaint:    BudgetCategoryMaterial,
	MaterialCategoryLighting: BudgetCategoryMaterial,
	MaterialCategoryBathroom: BudgetCategoryMaterial,
	MaterialCategoryHardware: BudgetCategoryMaterial,
	MaterialCategoryBoard:    BudgetCategoryMaterial,
	MaterialCategoryOther:    BudgetCategoryOther,
}

// PurchaseStatusOrder 采购状态推进顺序，值越大越靠后。
var PurchaseStatusOrder = map[string]int{
	PurchaseStatusNotPurchased: 0,
	PurchaseStatusOrdered:      1,
	PurchaseStatusDelivered:    2,
	PurchaseStatusInstalled:    3,
}

// BudgetCategoryForMaterial 返回材料品类对应的预算类别，未知品类返回 false。
func BudgetCategoryForMaterial(materialCategory string) (string, bool) {
	category, ok := MaterialCategoryBudgetMap[materialCategory]
	return category, ok
}

File: frontend/src/types/enums.ts
// 共享枚举：与后端 internal/constants/enums.go 保持一致。

export const DecorStyle = {
  Modern: 'Modern',
  Chinese: 'Chinese',
  Nordic: 'Nordic',
  Japanese: 'Japanese',
  Industrial: 'Industrial',
  Mediterranean: 'Mediterranean',
  Minimalist: 'Minimalist',
} as const
export type DecorStyle = (typeof DecorStyle)[keyof typeof DecorStyle]

export const PhaseStatus = {
  NotStarted: 'NotStarted',
  InProgress: 'InProgress',
  Revision: 'Revision',
  Approved: 'Approved',
} as const
export type PhaseStatus = (typeof PhaseStatus)[keyof typeof PhaseStatus]

export const ConstructionPhase = {
  Demolition: 'Demolition',
  Plumbing: 'Plumbing',
  Carpentry: 'Carpentry',
  Tiling: 'Tiling',
  Painting: 'Painting',
  Installation: 'Installation',
  SoftFurnishing: 'SoftFurnishing',
} as const
export type ConstructionPhase = (typeof ConstructionPhase)[keyof typeof ConstructionPhase]

export const PurchaseStatus = {
  NotPurchased: 'NotPurchased',
  Ordered: 'Ordered',
  Delivered: 'Delivered',
  Installed: 'Installed',
} as const
export type PurchaseStatus = (typeof PurchaseStatus)[keyof typeof PurchaseStatus]

export const ProjectStatus = {
  Designing: 'Designing',
  Quoting: 'Quoting',
  InProgress: 'InProgress',
  Completed: 'Completed',
  Archived: 'Archived',
} as const
export type ProjectStatus = (typeof ProjectStatus)[keyof typeof ProjectStatus]

export const ConstructionStatus = {
  Pending: 'Pending',
  InProgress: 'InProgress',
  Completed: 'Completed',
  Delayed: 'Delayed',
} as const
export type ConstructionStatus = (typeof ConstructionStatus)[keyof typeof ConstructionStatus]

export const AcceptanceStatus = {
  Pending: 'Pending',
  Passed: 'Passed',
  Failed: 'Failed',
} as const
export type AcceptanceStatus = (typeof AcceptanceStatus)[keyof typeof AcceptanceStatus]

export const Role = {
  Admin: 'Admin',
  Designer: 'Designer',
  Contractor: 'Contractor',
  Owner: 'Owner',
  ProjectManager: 'ProjectManager',
} as const
export type Role = (typeof Role)[keyof typeof Role]

export const HouseType = ['一室', '两室', '三室', '四室', '别墅', '复式'] as const
export type HouseType = (typeof HouseType)[number]

export const DesignPhaseName = ['方案设计', '效果图', '施工图', '软装方案'] as const
export const MaterialCategory = ['瓷砖', '地板', '油漆', '灯具', '卫浴', '五金', '板材', '其他'] as const
export const MaterialSpace = ['客厅', '卧室', '厨房', '卫生间', '阳台'] as const
export const BudgetCategory = ['Design', 'Material', 'Labor', 'Furniture', 'Appliance', 'Other'] as const
export type BudgetCategory = (typeof BudgetCategory)[number]

// 材料品类 -> 预算类别映射：与后端 internal/constants/enums.go 保持一致。
// 材料从已订货推进到已交付时，总价按项目 + 映射后的预算类别计入实际花费。
export const MaterialCategoryBudgetMap: Record<string, BudgetCategory> = {
  瓷砖: 'Material',
  地板: 'Material',
  油漆: 'Material',
  灯具: 'Material',
  卫浴: 'Material',
  五金: 'Material',
  板材: 'Material',
  其他: 'Other',
}

export function budgetCategoryForMaterial(category: string): BudgetCategory {
  return MaterialCategoryBudgetMap[category] ?? 'Other'
}
export const ConstructionName = ['拆改', '水电', '木工', '瓦工', '油漆', '安装', '软装'] as const
